### PR TITLE
Generating changelog using gren

### PR DESCRIPTION
**Pull request by gugu**
We can get it here: https://github.com/github-tools/github-release-notes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,581 @@
+# Changelog
+
+## v2.0.3 (01/01/1970)
```
Comment (cjbarth) [inline CHANGELOG.md:3]: This date is wrong

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+- [**closed**] normalize signature line endings before loading signature block to xml-crypto [#512](https://github.com/node-saml/passport-saml/pull/512)
```
Comment (cjbarth) [inline CHANGELOG.md:4]: Do we have to have the status here? Some of these status and tags really add nothing to the change log IMO.

Comment (cjbarth) [reply CHANGELOG.md:4]: Maybe we can use this config value to remove the tags: https://github-tools.github.io/github-release-notes/options.html#ignore-labels

File: .grenrc.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (cjbarth) [inline .grenrc.js:1]: It seems we might benefit from quickly tagging recent commits so we can use grouping. Thoughts? https://github-tools.github.io/github-release-notes/options.html#group-by-1

**Review by cjbarth — CHANGES_REQUESTED** [review]
Looks really nice. This should speed the process going forward. I see we can even put a badge if we wanted! https://github.com/github-tools/github-release-notes#setup

**Comment by gugu** [comment]
@cjbarth can you please look changelog one more time? Currently, it includes correct changes

File: CHANGELOG.md (hunk 2 above)
Comment (cjbarth) [reply CHANGELOG.md:4]: Can the date format be ISO 8601 standard? E.g. 2021-01-07

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+- [**closed**] Fix broken tests [#367](https://github.com/node-saml/passport-saml/pull/367)
+- [**peer-review-welcome**] Create a way to get provider metadata when using the MultiSamlStrategy [#323](https://github.com/node-saml/passport-saml/pull/323)
+- [**closed**] Extend and document the profile object [#301](https://github.com/node-saml/passport-saml/pull/301)
+- [**pending-refinement**] feat: add RequestedAuthnContext Comparison Type parameter [#360](https://github.com/node-saml/passport-saml/pull/360)
+- [**closed**] Update README.md [#363](https://github.com/node-saml/passport-saml/pull/363)
+- [**peer-review-welcome**] InResponseTo support for logout [#356](https://github.com/node-saml/passport-saml/pull/356)
+
+---
+
+## v0.32.0 (01/01/1970)
```
Comment (cjbarth) [inline CHANGELOG.md:385]: This seems both out-of-place and wrong.

File: CHANGELOG.md (hunk 2 above)
Comment (gugu) [reply CHANGELOG.md:4]: it will require custom template with date parsing/formatting code, there is no option for date format

Comment (cjbarth) [reply CHANGELOG.md:9]: It seems that way, sad... thankfully, someone already wrote the template, and all we have to do is paste it in :)

https://github.com/github-tools/github-release-notes/issues/147#issuecomment-584759005

File: .grenrc.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    "dataSource": "prs",
+    "prefix": "",
+    "onlyMilestones": false,
+    "ignoreTagsWith": ["v0.32.0", "0.0.3"],
```
Comment (cjbarth) [inline .grenrc.js:5]: Why are these ignored? Is that just for immediate expediency? We can always add a tag and then tell it to only work with tags with a "v" prefix if that helps.

Comment (gugu) [reply .grenrc.js:5]: These tags were created but no release was published. Changelog for these tags will be broken

Comment (gugu) [reply .grenrc.js:5]: and there is no chance to fix it because creating releases for old tags is not possible

File: CHANGELOG.md (hunk 2 above)
Comment (gugu) [reply CHANGELOG.md:9]: image

File: .grenrc.js (hunk 3 above)
Comment (gugu) [reply .grenrc.js:1]: I tried, but usually we have up to 10 PRs per release, grouping creates a group per 1-2 PRs

Comment (cjbarth) [reply .grenrc.js:1]: We can always revisit this later too.

**Comment by cjbarth** [comment]
Final question: is the script in `package.json` the one that generates this `CHANGELOG.md`? If not, could you please put the script you use there so that others can duplicate your work later on?

**Comment by cjbarth** [comment]
I'm not sure what I'm doing wrong then, because even that change you've made isn't working for me. I'm getting the correct tags, but the date is always 1969-12-31 and the message is always "*No changelog for this release.*". Given that we already have `tags: "all"` in the config file and `-G` is the same as `--generate`, I feel like something else is amiss here. You have this all working properly with `npm run changelog`?

**Review by cjbarth — APPROVED** [review]

**Comment by cjbarth** [comment]
I got it working reliably. I also use `prettier` on it, as I'd like to roll that out soon as we decide on the async/await thing; I don't want to break the merge too badly by using `prettier` before then. Feel free to merge this if you are happy with what I've done. Thanks for your hard work on this.

**Pull request merged**